rpc/transaction_test.go

```diff
@@ -261,154 +248,33 @@ func TestTransactionReceipt_MatchesCapturedTransaction(t *testing.T) {
-			t.Fatal("transaction receipt should exist")
-		}
-		txnReceipt, ok := txReceiptInterface.(InvokeTransactionReceipt)
-		if !ok {
-			t.Fatalf("transaction receipt should be InvokeTransactionReceipt, instead %T", txReceiptInterface)
-		}
-		if ok, err := regexp.MatchString(test.ExecutionStatus, string(txnReceipt.ExecutionStatus)); err != nil || !ok {
-			t.Fatal("error checking transaction status", ok, err, txnReceipt.ExecutionStatus)
-		}
-	}
-}
-
-// TestDeployOrDeclareReceipt is a test function that verifies the functionality of the DeployOrDeclareReceipt method.
-//
-// It tests the behavior of the DeployOrDeclareReceipt method by creating a test configuration, defining a test set,
-// and executing the test set for the specified test environment. It makes assertions to ensure that the expected
-// transaction receipt matches the actual transaction receipt returned by the method.
-//
-// The function takes no parameters and does not return any values.
-//
-// Parameters:
-// - t: the testing object for running the test cases
-// Returns:
-//
-//	none
-func TestDeployOrDeclareReceipt(t *testing.T) {
-	testConfig := beforeEach(t)
-
-	type testSetType struct {
-		TxnHash            *felt.Felt
-		ExpectedTxnReceipt TransactionReceipt
-	}
-
-	var receiptTxn300114_3 = DeclareTransactionReceipt(
-		CommonTransactionReceipt{
-			TransactionHash:    utils.TestHexToFelt(t, "0x46a9f52a96b2d226407929e04cb02507e531f7c78b9196fc8c910351d8c33f3"),
-			ActualFee:          FeePayment{Amount: utils.TestHexToFelt(t, "0x0"), Unit: UnitWei},
-			FinalityStatus:     TxnFinalityStatusAcceptedOnL1,
-			ExecutionStatus:    TxnExecutionStatusSUCCEEDED,
-			BlockHash:          utils.TestHexToFelt(t, "0x184268bfbce24766fa53b65c9c8b30b295e145e8281d543a015b46308e27fdf"),
-			BlockNumber:        300114,
-			Type:               "DECLARE",
-			MessagesSent:       []MsgToL1{},
-			Events:             []Event{},
-			ExecutionResources: ExecutionResources{Steps: 0},
-		})
-
-	testSet := map[string][]testSetType{
-		"mock": {},
-		"testnet": {
-			{
-				TxnHash:            utils.TestHexToFelt(t, "0x46a9f52a96b2d226407929e04cb02507e531f7c78b9196fc8c910351d8c33f3"),
-				ExpectedTxnReceipt: receiptTxn300114_3,
-			},
-		},
-		"mainnet": {},
-	}[testEnv]
-
-	for _, test := range testSet {
-		spy := NewSpy(testConfig.provider.c)
-		testConfig.provider.c = spy
-		txReceiptInterface, err := testConfig.provider.TransactionReceipt(context.Background(), test.TxnHash)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if txReceiptInterface == nil {
-			t.Fatal("transaction receipt should exist")
-		}
-		txnDeclareReceipt, ok := txReceiptInterface.(DeclareTransactionReceipt)
-		if !ok {
-			t.Fatalf("transaction receipt should be Deploy or Declare, instead %T", txReceiptInterface)
-		}
-		if !cmp.Equal(test.ExpectedTxnReceipt, txnDeclareReceipt) {
-			t.Fatalf("the expected transaction blocks to match, instead: %s", cmp.Diff(test.ExpectedTxnReceipt, txnDeclareReceipt))
-		}
```

Comment:
I am not sure why you are deleting these

Reply:
I figured having (and maintaining) a test for each Receipt type might be overkill. I can add them back in (and update them) if you think it's necessary

Reply:
We do have a test for TransactionReceipt in TestTransactionReceipt